service/src/builder.rs

```diff
@@ -69,22 +67,63 @@ fn extract_shuttle_toml_name(path: PathBuf) -> anyhow::Result<String> {
 pub async fn build_workspace(
     project_path: &Path,
     release_mode: bool,
-    tx: Sender<Message>,
+    tx: tokio::sync::mpsc::Sender<String>,
     deployment: bool,
 ) -> anyhow::Result<Vec<BuiltService>> {
     let project_path = project_path.to_owned();
-
     let manifest_path = project_path.join("Cargo.toml");
-
     if !manifest_path.exists() {
-        bail!(
-            "failed to read the Shuttle project manifest: {}",
-            manifest_path.display()
-        );
+        bail!("Cargo manifest file not found: {}", manifest_path.display());
     }
-    let metadata = cargo_metadata::MetadataCommand::new()
-        .manifest_path(&manifest_path)
-        .exec()?;
+
+    // Cargo's "Downloading ..." lines are quite verbose.
+    // Instead, a custom message is printed if the download takes significant time.
+    let mut command = tokio::process::Command::new("cargo");
+    command
+        .arg("fetch")
+        .arg("--manifest-path")
+        .arg(&manifest_path)
+        .arg("--color=always")
+        .stdout(Stdio::null())
+        .stderr(Stdio::null());
+    let notification = tokio::spawn({
+        let tx = tx.clone();
+        async move {
+            tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
```

Comment:
Why the wait?

Reply:
To mimic cargo fetch: skip printing this line if there is nothing to download. Only if it takes more than 500ms do we assume a download is happening and print this.